### PR TITLE
Adding initial VW auto-experimentation 

### PR DESCRIPTION
**Pull request by GalOshri**
Take raw JSON file from Decision Service and try to find good parameters. Runs a set of tests at a time to identify a good learning rate, good cb_type, good feature interactions, etc. 

Writes experiment results to file.

Various TODOs remaining to run better experiments.

### CODE REVIEW

File: DataScience/experimenter.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                shared_features.append(feature[0])
+        action_set = event['_multi']
+        for action in action_set:
+            for feature in action.keys():
+                if feature[0] != '_' and feature[0] not in action_features:
+                    action_features.append(feature[0])
+                    if action[feature].get('constant', 0) == 1 and 'id' in action[feature]:
+                        marginal_features.append(feature[0])
+        if counter >= 50:
+            break
```
Comment (eisber) [inline DataScience/experimenter.py]: why break here?

File: DataScience/experimenter.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    writer_p.start()        
+    
+    # Base parameters and setting up the cache file
+    base_command = "vw --cb_adf -d %s --json -c --power_t 0" % sys.argv[1] # TODO: VW location should be a command line parameter. 
+    initial_command = Command(base_command, learning_rate=0.5)
+    run_experiment(initial_command, timeout=3600)
+    
+    # Learning rates
+    command_list = []
+    learning_rates = [0.5, 0.25, 0.1, 0.05, 0.025, 0.01, 0.005, 0.0025, 0.001, 0.0005, 0.00025, 0.0001, 0.00005, 0.000025, 0.00001]
```
Comment (eisber) [inline DataScience/experimenter.py]: externalize to ini/config file? the ini module I used for Azure download was straight to use

File: DataScience/experimenter.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        
+        results = run_experiment_set(command_list)
+        if results[0].loss < best_loss:
+            best_loss = results[0].loss
+            best_interaction_list = list(results[0].interaction_list)
+        else:
+            break
+        
+    # Regularization
+    regularizations = [1e-1, 1e-2, 1e-3, 1e-4, 1e-5, 1e-6, 1e-7, 1e-8, 1e-9, ""]
```
Comment (eisber) [inline DataScience/experimenter.py]: externalize?

**Comment by eisber** [comment]
nice work

File: DataScience/experimenter.py (hunk 1 above)
Comment (GalOshri) [inline DataScience/experimenter.py]: We assume schema is fixed so we only need to read a few lines to capture all the features. I added a comment.
